### PR TITLE
Open project in IDE

### PR DESCRIPTION
**Pull request by orimdominic**
This PR affects issue #1561 

I've been able to resolve this with a few caveats.

If no code editor is open, it will try to open it in VS Code by default. This happens because [guessEditor](https://github.com/remotion-dev/remotion/blob/a255b47c0daa632e4fbb49675853dff618a42788/packages/cli/src/preview-server/error-overlay/react-overlay/utils/open-in-editor.ts#L346) returns the currently open code editor and when there is none, it returns undefined

PS: I noticed that when Sublime Text is open on my computer and it should be the editor that opens up the project, I get the error in the screenshot below when I try to open the project.

Screenshot from 2022-12-14 11-47-14

### CODE REVIEW

File: packages/cli/src/editor/helpers/use-menu-structure.tsx [hunk 1]
```diff
@@ -151,6 +151,34 @@ export const useMenuStructure = (closeMenu: () => void) => {
 						subMenu: null,
 						quickSwitcherLabel: 'New still...',
 					},
+					{
```
Comment (JonnyBurger) [inline packages/cli/src/editor/helpers/use-menu-structure.tsx]: The action should only be displayed if `window.remotion_editorName` is not null

File: packages/cli/src/editor/helpers/use-menu-structure.tsx [hunk 2]
```diff
@@ -151,6 +151,34 @@ export const useMenuStructure = (closeMenu: () => void) => {
...
+						id: 'open-in-editor',
+						value: 'open-in-editor',
+						label: 'Open in code editor...',
```
Comment (JonnyBurger) [inline packages/cli/src/editor/helpers/use-menu-structure.tsx]: Can we rename it to `Open in ${window.remotion_editorName}`?
The `...` signals that a modal is opened in the Preview, so I would remove the `...`.

File: packages/cli/src/editor/helpers/use-menu-structure.tsx [hunk 3]
```diff
@@ -151,6 +151,34 @@ export const useMenuStructure = (closeMenu: () => void) => {
...
+						onClick: async () => {
+							await fetch(`/api/open-in-editor`, {
```
Comment (JonnyBurger) [inline packages/cli/src/editor/helpers/use-menu-structure.tsx]: Now duplicated with the code in `OpenInEditor`, can we create a helper function to be used in both places?

File: packages/cli/src/editor/helpers/use-menu-structure.tsx [hunk 4]
```diff
@@ -151,6 +151,34 @@ export const useMenuStructure = (closeMenu: () => void) => {
...
+								body: JSON.stringify({
+									stack: {
+										originalFileName: `${window.remotion_cwd}`,
+										originalLineNumber: 1,
+										originalColumnNumber: 1,
+									},
+								}),
+							})
+							.catch(err => {
+								console.error(err)
```
Comment (JonnyBurger) [inline packages/cli/src/editor/helpers/use-menu-structure.tsx]: You can handle the error with `notificationCenter.current?.addNotification()`.

File: packages/cli/src/preview-server/routes.ts [hunk 5]
```diff
@@ -148,9 +148,10 @@ const handleOpenInEditor = async (
 		const stack = body.stack as SymbolicatedStackFrame;
 
 		const guess = await editorGuess;
+		const defaultCodeEditor = { process: 'code', command: 'code' }
 		const didOpen = await launchEditor({
 			colNumber: stack.originalColumnNumber as number,
-			editor: guess[0],
```
Comment (JonnyBurger) [inline packages/cli/src/preview-server/routes.ts]: Let's not default to VS Code, instead if the editor was not guessed, let's not display the menu at all.

**Review by JonnyBurger — CHANGES_REQUESTED** [review]
Thanks a lot! 🙌
It works, some polishing needed.

**Comment by JonnyBurger** [comment]
How do you launch Sublime normally? Should we rename `sublime_text` to `sublime`?

**Comment by JonnyBurger** [comment]
Alright, thanks!
And sorry for pushing a commit to this branch, was accidentially. Will be in main soon so the diff will be clean afterwards.

**Comment by JonnyBurger** [comment]
Awesome! Looks great, thanks a lot! 🙌

@MehmetAdemi Can you reward the issue?¨

**Comment by MehmetAdemi** [comment]
For sure! 
Thanks @orimdominic! Can you go to issue #1561 and sign up on Algora so I can make the payment?

**Review by JonnyBurger — APPROVED** [review]

**Comment by orimdominic** [comment]
> For sure @JonnyBurger ! Thanks @orimdominic! Can you go to issue #1561 and sign up on Algora so I can make the payment?

I will soonest!

**Comment by orimdominic** [comment]
Felt great contributing to this!

**Comment by JonnyBurger** [comment]
Awesome! Your contribution is much appreciated 🙌 💙

**Pull request merged**